core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -227,6 +228,27 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       val tableTypes = hiveTables.map(ht =>
         Utils.toUpperCase(s"${ht.schema}.${ht.entityName}") -> ht.tableType).toMap
       val regionStats = result.flatMap(_.getRegionStats.asScala).map(rs => {
+        val tableRegion = Misc.getRegionForTable(rs.getTableName, false)
+        if (tableRegion != null && tableRegion.isInstanceOf[PartitionedRegion]) {
+          val tablePrRegion = tableRegion.asInstanceOf[PartitionedRegion]
+          val PrRegRedProvider = tablePrRegion.getRedundancyProvider
+          if (PrRegRedProvider != null) {
+            rs.setRedundancyImpaired(PrRegRedProvider.isRedundancyImpaired)
+            rs.setRedundancy(tablePrRegion.getRedundantCopies)
+          }
+
+          val numBuckets = tablePrRegion.getPartitionAttributes.getTotalNumBuckets
+          breakable {
+            for (i <- 0 until numBuckets) {
+              val idm = tablePrRegion.getNodeForBucketRead(i)
```

Comment:
Test this for time. Use a schema which has multiple tables with a lot of buckets and please check the time it takes to complete all calls. Since this is going to be called for every run so we should check the cost of this.
For example take at least 100 tables with default number of bucket which means 12800 calls will be done.

Reply:
Acknowledged, will check it and will enhance the code in next release.